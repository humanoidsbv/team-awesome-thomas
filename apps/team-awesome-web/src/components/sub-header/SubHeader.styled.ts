import styled from "styled-components";

export const SubHeader = styled.div`
  align-content: center;
  background-color: ${({ theme }) => theme.backgroundPrimary};
  border-bottom: solid 1px ${({ theme }) => theme.grey3};
  display: grid;
  gap: 20px;
  grid-auto-flow: row;
  padding: 20px;
  width: 100%;

  @media screen and (${({ theme }) => theme.tablet}) {
    grid-auto-flow: column;
    justify-content: space-between;
    padding: 16px 32px;
  }
`;

export const ContextMenu = styled.div`
  display: grid;
  gap: 20px;
  grid-auto-flow: column;
  justify-content: space-between;
  width: 100%;

  & h1,
  & p {
    align-content: center;
    display: grid;
    font-weight: ${({ theme }) => theme.fontWeightBold};
    grid-auto-flow: column;
    white-space: nowrap;
  }

  @media screen and (${({ theme }) => theme.tablet}) {
    width: auto;

    & h1::after {
      align-self: center;
      border-left: solid 1px ${({ theme }) => theme.grey5};
      content: "";
      height: ${({ theme }) => theme.fontSizeMedium};
      margin-left: 20px;
    }
  }
`;

export const ContextHeading = styled.h1`
  color: ${({ theme }) => theme.grey6};
  font-family: ${({ theme }) => theme.fontPrimary};
  font-size: 18px;
`;

export const ContextCount = styled.p`
  color: ${({ theme }) => theme.grey5};
  font-family: ${({ theme }) => theme.fontPrimary};
  font-size: ${({ theme }) => theme.fontSizeMedium};
  margin-left: auto;
`;
