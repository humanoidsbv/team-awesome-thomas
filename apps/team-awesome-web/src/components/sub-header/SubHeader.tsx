import * as Styled from "./SubHeader.styled";
import { Button } from "../button/";

export const SubHeader = () => {
  return (
    <Styled.SubHeader>
      <Styled.ContextMenu>
        <Styled.ContextHeading>Team members</Styled.ContextHeading>
        <Styled.ContextIndicator>22 Humanoids</Styled.ContextIndicator>
      </Styled.ContextMenu>
      <Button label="New Humanoid" href="treinen.online/" />
    </Styled.SubHeader>
  );
};
