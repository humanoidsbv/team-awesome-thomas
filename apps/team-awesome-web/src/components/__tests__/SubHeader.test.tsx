import React from "react";
import renderer from "react-test-renderer";
import { SubHeader } from "../sub-header";

it("Subheader renders correctly", () => {
  const tree = renderer.create(<SubHeader count="420" title="test" />).toJSON();
  expect(tree).toMatchSnapshot();
});
