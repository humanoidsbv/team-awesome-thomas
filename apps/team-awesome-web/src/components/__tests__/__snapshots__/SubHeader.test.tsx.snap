// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subheader renders correctly 1`] = `
<div
  className="sc-bcXHqe fGUCmM"
>
  <div
    className="sc-gswNZR jxYIvT"
  >
    <h1
      className="sc-dkrFOg ftkZir"
    >
      test
    </h1>
    <p
      className="sc-hLBbgP dNcsSF"
    >
      420
    </p>
  </div>
</div>
`;
